### ISSUE TITLE
`out_of_slice_display` naming inconsistent between Qt controls and `Points`/`Vectors` layers

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#4028]
In #4007 `n_dimensional` was refactored to `out_of_slice_display`. The Qt controls for this are named as if the attribute were callled `out_of_slice` - we should fix this inconsistency to avoid any future confusion

cc @brisvag

**Comment by brisvag** [comment]
Since it was inconsisent before (`n_dimensional` in the code, `ndim` in the gui) I figured it was fine. I think it's more readable as `out_of_slice` in the gui, but I don't feel strongly about this, it's probably better to go for consistency!

**Comment by psobolewskiPhD** [comment]
Just for clarity, the issue is the GUI display here:
<img width="301" alt="image" src="https://github.com/user-attachments/assets/6d2b49b8-0819-405d-8ab9-f1fcecf6b1bd">

I tend to think `out of slice` isn't very obvious what it means vs. `out of slice display`.
The tool tip could also be updated to be a bit more helpful.

**Comment by brisvag** [comment]
Good point... Yeah this can be implemented as a different projection mode, where size gets interpolated between max and zero as you approach the edge of the thick slice.

**Comment by DragaDoncila** [comment]
Ok so I guess to fully deprecate the checkbox we should implement the scaled projection mode for points first
